src/error.rs

```diff
@@ -324,20 +327,32 @@ impl From<LzwError> for TiffError {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+struct JpegDecoderError {
+    inner: jpeg::Error,
+}
+
+impl From<jpeg::Error> for JpegDecoderError {
+    fn from(error: jpeg::Error) -> Self {
+        Self { inner: error }
+    }
+}
```

Comment:
Please make sure to remove this `impl` and construct the error with a private converter. Otherwise, upgrading the `jpeg_decoder` dependency is a SemVer visible detail.